js/holidays.js

```diff
@@ -354,51 +354,76 @@ function getMonthDays( year ) {
 }
 
 /**
- * Calculates the observation date for a given holiday according to a country's rules
+ * Returns an array of actual and observed holiday dates according to a country's rules
  *
- * @param {number} year
- * @param {number} month
- * @param {number} day
- * @param {string} country
+ * I.e. if a holiday falls on a Sunday and an Monday becomes a public holiday instead,
+ * both Sunday and Monday dates will be returned.
  *
- * @returns {string} holiday date in 'month-day' format
+ * @param {number} year Holiday year
+ * @param {number} month Holiday month
+ * @param {number} day Holiday day
+ * @param {string} country Country code
+ * @param {string} name Holiday name
+ * @param {Object} options Additional options
+ * @param {string} options.observedName Holiday name for the observed date
+ * @param {number} options.consecutive How many consecutive holidays is this one a part of (to avoid shifting observed date onto them)
  */
-function calcObservation( year, month, day, country ) {
+function observed( year, month, day, country, name, options ) {
+	options = options || {};
 
-	var diff = 0,
+	var diffs = [0],
 		date = new Date( year, month - 1, day ),
 		dow = date.getDay();
 
 	switch ( country ) {
 		case 'mx' :
 			if ( dow == 0 )
-				diff = 1;
+				diffs = [1];
 			else if ( dow == 6 )
-				diff = -1;
+				diffs = [-1];
 			break;
 
 		case 'au' :
 		case 'uk' :
+			var consecutive = options.consecutive || 1;
 			if ( dow == 0 )
-				diff = 1;
+				diffs.push( consecutive );
 			else if ( dow == 6 )
-				diff = 2;
+				diffs.push( 2 );
 			break;
 
 		case 'ar' :
 		case 'uy' :
 			if ( dow == 2 )
-				diff = -1;
+				diffs = [-1];
 			else if ( dow == 3 )
-				diff = -2;
+				diffs = [-2];
 			else if ( dow == 4 )
-				diff = 4;
+				diffs = [4];
 			else if ( dow == 5 )
-				diff = 3;
+				diffs = [3];
 			break;
 	}
 
-	return dateToMonthDay( dateAdd( date, diff ) );
+	var observedName = options.observedName;
+	if ( !observedName ) {
+		switch ( country ) {
+			case 'au':
+				observedName = `Additional public holiday for ${name}`;
+				break;
+			case 'uk':
+				observedName = `${name} (in lieu)`;
+				break;
+		}
+	}
```

Comment:
Do you think we could simplify this by just adding "(observed)" to the original holiday name, as you proposed in #7?
The actual text could be a string in the translations object and we would do it for all countries which have observed holidays.

Reply:
I went by the official names for [Australia](https://www.fairwork.gov.au/employment-conditions/public-holidays/2023-public-holidays). UK seems to [disagree](https://www.gov.uk/bank-holidays); I'm happy to put "in lieu" for both.

However, the holiday names themselves aren't translated, so it doesn't make sense to translate "in lieu". (Does "Año Nuevo (in lieu)" look good?)

Reply:
From a quick look, doesn't seem like there is an official spelling. Will you be happy with "(in lieu)" added only in AU and UK?

Reply:
Both AU and UK seem to use "substitute day" in those pages. I remember seeing "observed" in US calendars. 
You have a point with the holiday names, but "Año Nuevo (in lieu)" looks weird to me. I think we either use the same language of the holiday name or the language selected for the calendar.

Reply:
> From a quick look, doesn't seem like there is an official spelling. Will you be happy with "(in lieu)" added only in AU and UK?

I think we can do this for now.

Reply:
All fixed.